### PR TITLE
Use and enforce Astropy compliant units

### PR DESCRIPTION
**Pull request by pschella**

### CODE REVIEW

File: python/lsst/meas/algorithms/loadReferenceObjects.py [hunk 1]
```diff
@@ -306,15 +306,15 @@ def makeMinimalSchema(filterNameList, addFluxSigma=False,
...
+                units = "count",
             )
         if addFluxSigma:
             for filterName in filterNameList:
                 schema.addField(
                     field = "%s_fluxSigma" % (filterName,),
                     type = numpy.float64,
                     doc = "flux uncertainty in filter %s" % (filterName,),
-                    units = "?",
+                    units = "count",
```
Comment (TallJimbo) [inline python/lsst/meas/algorithms/loadReferenceObjects.py]: I don't think the units in reference catalogs are actually counts.  We had a long debate on what they should be, but I don't remember the outcome; maybe @r-owen does?

Comment (timj) [inline python/lsst/meas/algorithms/loadReferenceObjects.py]: All your options are listed in http://www.ivoa.net/documents/VOUnits/20140523/VOUnits-REC-1.0-20140523.pdf

Comment (TallJimbo) [inline python/lsst/meas/algorithms/loadReferenceObjects.py]: Well we're using *some* unit already, and I'm not proposing we change it right now if it's not compliant (but it's much better to leave it blank than set it incorrectly).

Comment (r-owen) [inline python/lsst/meas/algorithms/loadReferenceObjects.py]: The units are Janskys, as per http://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/classlsst_1_1meas_1_1algorithms_1_1load_reference_objects_1_1_load_reference_objects_task.html#meas_algorithms_loadReferenceObjects_Schema

**Pull request merged**